sdk/core/Azure.Core/tests/BearerTokenAuthenticationPolicyTests.cs

```diff
@@ -366,32 +367,44 @@ public void BearerTokenAuthenticationPolicy_OneHundredConcurrentCallsFailed()
         }
 
         [Test]
-        public async Task BearerTokenAuthenticationPolicy_GatedConcurrentCallsFailed()
+        public void BearerTokenAuthenticationPolicy_GatedConcurrentCallsFailed()
```

Comment:
Now would be a good time to add some comments to this test to describe what sequence of events and what behavior is expected here. 😄 Some of these tests are a bit difficult to grok at first glance.